app/Notifications/PostTagged.php

```diff
@@ -71,52 +85,36 @@ public function via($notifiable)
     }
 
     /**
-     * Get the mail representation of the notification.
+     * Get the Slack representation of the notification.
      *
-     * @param  mixed  $notifiable
-     * @return \Illuminate\Notifications\Messages\MailMessage
+     * @param mixed $notifiable
+     * @return \Illuminate\Notifications\Messages\SlackMessage
      */
     public function toSlack($notifiable)
     {
-        if (! config('services.slack.url')) {
-            return;
-        }
-
-        // Get the user & admin reviewer's names for the notification.
-        $data = app(GraphQL::class)->query(SLACK_NOTIFICATION_QUERY, [
-            'userId' => $this->post->northstar_id,
-            'adminId' => $this->adminId,
-        ]);
+        $data = $this->queryForUserAndReviewer();
```

Comment:
No biggie, but why not just make the query here vs create another function?

Reply:
A ya, good question!

So for one, I just thought it reads a little better and looks a little clearer/cleaner. When reviewing the `toSlack()` method you don't really need to worry about seeing the call to `GraphQL` class and `query()` with the constant passed to it etc. All you need to grok is that you're getting `$data` and that data is from a "queryForUserAndReviewer". If you care to know what's involved in that you can check the `queryForUserAndReviewer()`.

But that's more an opinionated reason. For a more practical reason, this `Notification` class is not just a Slack notification class (which is why I changed the name), but meant to potentially notify a number of channels after a post is tagged. 

We happen to only use the `toSlack()` method, but if we did decide we also want to maybe email the marketing team the posts tagged as "good", or also log to database, we could add those channels in this class as well. They would each have their own `to*()` methods, and they'd have access to calling `$this->queryForUserAndReviewer()` to get that same data, instead of having to repeat it in each method.